### ISSUE TITLE
TIFFImageWriter does not correctly write images with "sampleModelTranslate"

### ISSUE THREAD

**Issue by haraldk** [haraldk/TwelveMonkeys#228]
TIFFImageWriter does not correctly write images with "sampleModelTranslate" (ie. as returned by `BufferedImage.getSubimage(...)`). 

The writer is accessing the pixel data directly without taking the offset and the original pixel stride into account, and as a result, the written image looked skewed (or in other words, just completely messed up).

See #226 for background & discussion.

**Comment by haraldk** [comment]
Failing test case for the `TIFFImageWriterTest`:

````java

    @Test
    public void testWriteCropped() throws IOException, URISyntaxException {
        // TODO: Test 2, 4, 8 bit palette data + gray + ushort + float types?
        List<URL> testData = Arrays.asList(
                getClass().getResource("/tiff/quad-lzw.tif"),
                getClass().getResource("/tiff/ccitt/group3_1d.tif")
        );

        for (URL resource : testData) {
            // Read it
            BufferedImage original = ImageIO.read(resource);

            // Crop it
            BufferedImage subimage = original.getSubimage(original.getWidth() / 4, original.getHeight() / 4, original.getWidth() / 2, original.getHeight() / 2);

            // Store cropped
            ByteArrayOutputStream bytes = new ByteArrayOutputStream();
            try (ImageOutputStream output = ImageIO.createImageOutputStream(bytes)) {
                ImageWriter imageWriter = createImageWriter();
                imageWriter.setOutput(output);
                imageWriter.write(subimage);
            }

            // Re-read cropped
            BufferedImage cropped = ImageIO.read(new ByteArrayImageInputStream(bytes.toByteArray()));

            // Compare
            assertImageEquals(String.format("Error comparing cropped to output: %s", new File(resource.toURI()).getName()), subimage, cropped);
        }
    }

    private void assertImageEquals(final String message, final BufferedImage expected, final BufferedImage actual) {
        assertNotNull(message, expected);
        assertNotNull(message, actual);
        assertEquals(message + ", widths differ", expected.getWidth(), actual.getWidth());
        assertEquals(message + ", heights differ", expected.getHeight(), actual.getHeight());

        for (int y = 0; y < expected.getHeight(); y++) {
            for (int x = 0; x < expected.getWidth(); x++) {
                assertEquals(String.format("%s, ARGB differs at (%s,%s)", message, x, y), expected.getRGB(x, y), actual.getRGB(x, y));
            }
        }
    }
````

**Comment by haraldk** [comment]
Fixed in c188931.

-- 
Harald K
